src/controllers/module/event/eventController.ts

```diff
@@ -96,29 +99,25 @@ export class EventController {
 			errors.size = Object.keys(errors.fields).length
 
 			let event = await this.eventFactory.create(data)
-			if (eventId) {
-				event = response.locals.event
-			}
 
 			if (errors.size) {
 				request.session!.sessionFlash = {errors: errors, event: event}
-				return response.redirect(`/content-management/courses/${courseId}/modules/${moduleId}/events/`)
-			}
-
-			let savedEvent
-			if (eventId) {
-				event.dateRanges.push(data.dateRanges[0])
-
-				savedEvent = await this.learningCatalogue.updateEvent(courseId, moduleId, eventId, event)
+				request.session!.save(() => {
+					response.redirect(`/content-management/courses/${courseId}/modules/${moduleId}/events`)
+				})
 			} else {
-				savedEvent = await this.learningCatalogue.createEvent(courseId, moduleId, event)
-			}
+				let eventToMerge = eventId ? response.locals.event : request.session!.event
 
-			request.session!.sessionFlash = {event: savedEvent}
+				if (eventToMerge) {
+					eventToMerge.dateRanges.push(event.dateRanges[0])
```

Comment:
It appears lines 114,115,191 have not been unit tested in this class. Can you check and update if necessary?

Reply:
yes ,addressed in the recent commit